MySQL: Cannot connect - "Name is null"

### Description

Currently I experience a problem I can't explain myself. I try to connect to a MySQL5 Server and always get the error 'Internal error connecting to <Connection Name> -  Name is null'.
I tried both variants, MySQL5 and MySQL using the latest connector library 8.1, but no success so far. Using the mysql client from the command line is no problem and works. The connection is pretty straight forward, just IP, port and username, no SSL or other things in use.
Any ideas what I could try?


### DBeaver Version

Community edition  23.2.0.202309031556

### Operating System

Ubuntu 20.04

### Database and driver

5.7.6 CData MySQL Remoting

MySQL Connector 5.1.44
MySQL Connector 8.0.29
MySQL Connector 8.1.0

### Steps to reproduce

1. Configured the database connection (regardless if option URL or Hostname)
2. Connect to database

### Additional context

None

Hi,
well, the output is not very promising:
```
org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [1003] [HY000]: The SQL statement 'SELECT [ENGINES].[SUPPORT] FROM [information_schema].[ENGINES] AS [ENGINES]' is not supported.
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:133)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:582)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$1(SQLQueryJob.java:491)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:190)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:498)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:934)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:3940)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:123)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:190)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:121)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:5142)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.sql.SQLException: The SQL statement 'SELECT [ENGINES].[SUPPORT] FROM [information_schema].[ENGINES] AS [ENGINES]' is not supported.
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:130)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
	at com.mysql.cj.jdbc.StatementImpl.executeInternal(StatementImpl.java:763)
	at com.mysql.cj.jdbc.StatementImpl.execute(StatementImpl.java:648)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:330)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	... 12 more
```
As mentioned earlier, it is not a real database, it's a connector acting like a MySQL DB. This piece of software is a translator between normal SQL and NoSQL and gets its data from a MongoDB.

Regards, Thomas

--- a/plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLEngine.java
+++ b/plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLEngine.java
@@ -21,6 +21,7 @@ import org.jkiss.code.Nullable;
 import org.jkiss.dbeaver.ext.mysql.MySQLConstants;
 import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
 import org.jkiss.dbeaver.model.meta.Property;
+import org.jkiss.utils.CommonUtils;
 
 import java.sql.ResultSet;
 import java.sql.SQLException;
@@ -63,7 +64,10 @@ public class MySQLEngine extends MySQLInformation {
     {
         this.name = JDBCUtils.safeGetString(dbResult, MySQLConstants.COL_ENGINE_NAME);
         this.description = JDBCUtils.safeGetString(dbResult, MySQLConstants.COL_ENGINE_DESCRIPTION);
-        this.support = Support.valueOf(JDBCUtils.safeGetString(dbResult, MySQLConstants.COL_ENGINE_SUPPORT));
+        this.support = CommonUtils.valueOf(
+            Support.class,
+            JDBCUtils.safeGetString(dbResult, MySQLConstants.COL_ENGINE_SUPPORT),
+            Support.YES);
         this.supportsTransactions = "YES".equals(JDBCUtils.safeGetString(dbResult, MySQLConstants.COL_ENGINE_SUPPORT_TXN));
         this.supportsXA = "YES".equals(JDBCUtils.safeGetString(dbResult, MySQLConstants.COL_ENGINE_SUPPORT_XA));
         this.supportsSavepoints = "YES".equals(JDBCUtils.safeGetString(dbResult, MySQLConstants.COL_ENGINE_SUPPORT_SAVEPOINTS));

